db/python/layers/ourdna/dashboard.py

```diff
@@ -15,93 +15,106 @@
-            sample=sample, property_name='process-end-time'
-        )
-        _processing_site = self.get_meta_property(
-            sample=sample, property_name='processing-site'
-        )
-        if (
-            _process_start_time is None
-            or _process_end_time is None
-            or _processing_site is None
-        ):
-            return None, None
-
-        processing_time = datetime.strptime(
-            _process_end_time, '%Y-%m-%d %H:%M:%S'
-        ) - datetime.strptime(_process_start_time, '%Y-%m-%d %H:%M:%S')
-
-        return _processing_site, int(processing_time.total_seconds())
-
-    def get_collection_to_process_start_time(self, sample: Sample) -> int | None:
-        """
-        I want to know how long it has been since the sample was collected - SAMPLE TABLE
-        @fields: collection-time, process-start-time
-        """
-        _collection_time = self.get_meta_property(
-            sample=sample, property_name='collection-time'
-        )
-        _process_start_time = self.get_meta_property(
-            sample=sample, property_name='process-start-time'
-        )
-        if _collection_time is None or _process_start_time is None:
-            return None
+    @staticmethod
+    def from_sample(sample: Sample) -> 'SampleProcessMeta':
+        """Create a SampleProcessMeta instance from a Sample instance."""
+        instance = SampleProcessMeta()
+        instance.meta = sample.meta
+        return instance
+
+    @property
+    def collection_to_process_end_time(self) -> int | None:
+        """Get the time taken from collection to process end."""
+        if self._collection_to_process_end_time is None:
+            self._collection_to_process_end_time = self.get_time_difference_in_seconds(
+                'collection-time', 'process-end-time'
+            )
+        return self._collection_to_process_end_time
+
+    @property
+    def processing_times_by_site(self) -> tuple[str | None, int | None]:
+        """Get processing times and site for a sample."""
+        if self._processing_times_by_site is None:
+            processing_site = self.get_property('processing-site')
+            processing_time = self.get_time_difference_in_seconds(
+                'process-start-time', 'process-end-time'
+            )
+            self._processing_times_by_site = (processing_site, processing_time)
+        return self._processing_times_by_site
+
+    @property
+    def collection_to_process_start_time(self) -> int | None:
+        """Get the time taken from collection to process start."""
+        if self._collection_to_process_start_time is None:
+            self._collection_to_process_start_time = (
+                self.get_time_difference_in_seconds(
+                    'collection-time', 'process-start-time'
+                )
+            )
+        return self._collection_to_process_start_time
+
+    @property
+    def get_lost_sample_properties(self) -> dict[str, Any]:
```

Comment:
If you store a reference to the sample, you can return a `OurDNALostSample` and get init checking through linting, make your life easy here!